backend/deploy.js

```diff
@@ -1,11 +1,12 @@
 var crypter = require("../util/crypter.js");
 var spawn = require('child_process').spawn;
 var output = require('../util/output');
+var socketHandler = require('../util/socketHandler.js');
 
-exports.deployPages = function (socket, data) {
+exports.deployPages = function (socket, data, callback) {
```

Comment:
adding callbacks to code just for the sake of testing seems like a bad idea. If the function doesn't return anything, try to test the side effects. Like if you want to test that a method is called, maybe mock that method using any mock testing framework.